libvast/src/system/exporter.cpp

```diff
@@ -154,11 +162,15 @@ behavior exporter(stateful_actor<exporter_state>* self, expression expr,
           self->state.results.push_back(std::move(candidate));
         else
           VAST_DEBUG(self, "ignores false positive:", candidate);
-        mask.append_bits(false, candidate.id() - mask.size());
-        mask.append_bit(true);
+        if (sender == self->state.index || sender == self->state.archive) {
+          mask.append_bits(false, candidate.id() - mask.size());
+          mask.append_bit(true);
+        }
+      }
+      if (sender == self->state.index || sender == self->state.archive) {
+        self->state.stats.processed += candidates.size();
+        self->state.unprocessed -= mask;
```

Comment:
> And the (un)processed candidates sounded like they relate to the protocol between EXPORTER and ARCHIVE / INDEX in those cases. If I was mistaken, I'll remove the if.

You're correct about the `unprocessed` bitmap. However, the `processed` count is independent of the historical query logic. They `unprocessed` bitmap is used to determine whether a historical query has completed: if there are still outstanding hits, the query cannot shutdown.

Let's keep the check. One thing I didn't notice earlier: change
```
if (sender == self->state.index || sender == self->state.archive)
```
to
```
if (self->current_sender() == self->state.archive)
```
because an INDEX will never send events (only bitmaps).